docs/topics/advent-of-code.md

```diff
@@ -0,0 +1,74 @@
+
+* Read the puzzle description on [Advent of Code](https://adventofcode.com/2020/day/1)
+* Check out the solution by Svetlana Isakova in [Kotlin Blog](https://blog.jetbrains.com/kotlin/2021/07/advent-of-code-in-idiomatic-kotlin/)
+or watch the video:
+
+<video width="560" height="315" href="o4emra1xm88" title="Kotlin Tutorial: Advent of Code Puzzles, Day 1"/>
+
+## Day 2. Password philosophy
+
+In this puzzle, you'll explore string utility functions, regular expressions, operations on collections,
+and how the `let` function can be helpful to transform your expressions.
+
+* Read the puzzle description on [Advent of Code](https://adventofcode.com/2020/day/2)
+* Check out the solution by Svetlana Isakova in [Kotlin Blog](https://blog.jetbrains.com/kotlin/2021/07/advent-of-code-in-idiomatic-kotlin-day2/)
+or watch the video:
+
+<video width="560" height="315" href="MyvJ7G6aErQ" title="Kotlin Tutorial: Advent of Code Puzzles, Day 2"/>
+
+## Day 3. Toboggan trajectory
+
+In this puzzle, you'll compare imperative and more functional code styles, work with pairs and the `reduce` function,
+edit code in column selection mode, and fix integer overflows.
+
+* Read the puzzle description on [Advent of Code](https://adventofcode.com/2020/day/3)
+* Check out the solution by Mikhail Dvorkin on [GitHub](https://github.com/kotlin-hands-on/advent-of-code-2020/blob/master/src/day03/day3.kt)
+or watch the video:
+
+<video width="560" height="315" href="ounCIclwOAw" title="Kotlin Tutorial: Adopting a Functional Style for Advent of Code Puzzles"/>
+
+## Day 4. Passport processing
+
+In this puzzle, you'll apply the `when` statement and explore different ways of how to validate the input — using
+utility functions, working ranges, checking set membership, and matching a particular regular expression.
+
+* Read the puzzle description on [Advent of Code](https://adventofcode.com/2020/day/4)
+* Check out the solution by Sebastian Aigner in [Kotlin Blog](https://blog.jetbrains.com/kotlin/2021/09/validating-input-advent-of-code-in-kotlin/) or watch the video:
+
+<video width="560" height="315" href="-kltG4Ztv1s" title="Kotlin Tutorial: Validating and Sanitizing Input. Advent of Code Puzzles"/>
+
+## Day 5. Binary boarding
+
+In this puzzle, you'll use the Kotlin standard library functions to work with the binary representation of numbers,
+explore powerful local functions, and learn how to use the `max` function in Kotlin 1.5 correctly.
+
+* Read the puzzle description on [Advent of Code](https://adventofcode.com/2020/day/5)
+* Check out the solution by Svetlana Isakova in [Kotlin Blog](https://blog.jetbrains.com/kotlin/2021/09/idiomatic-kotlin-binary-representation/)
+or watch the video:
+
+<video width="560" height="315" href="XEFna3xyxeY" title="Kotlin Tutorial: Binary Representation of Numbers. Advent of Code Puzzles"/> 
+
+## Day 6. Custom customs
+
+In this puzzle, you'll learn how to group and count characters in strings and collections using the standard library
```

Comment:
TBH, I don't like "In this puzzle" at every section, but I can't provide the better solution here.

Reply:
I'd agree. We could shorten them overall to "Learn how to group and count...", maybe.